src/runtime_src/core/tools/xbutil2/SubCmdStatus.cpp

```diff
@@ -0,0 +1,605 @@
+  //return max length of the cuname and port names
+  size_t max1 = 0, max2 = 0;
+  char sep = '/';
+  for (auto slotName: aSlotNames) {
+    size_t found1;
+    size_t start = 0;
+    found1 = slotName.find(sep, 0);
+    if (found1 == 0) {
+      //if the cuname starts with a '/'
+      start = 1;
+      found1 = slotName.find(sep, 1);
+    }
+    if (found1 != std::string::npos) {
+      aCUNamePortNames.emplace_back(slotName.substr(start, found1-start), slotName.substr(found1+1));
+    }
+    else {
+      aCUNamePortNames.emplace_back("Unknown", "Unknown");
+    }
+    //Replace the name of the host-AIM to something simple
+    if (aCUNamePortNames.back().first.find("interconnect_host_aximm") != std::string::npos) {
+      aCUNamePortNames.pop_back();
+      aCUNamePortNames.emplace_back("XDMA", "N/A");
+    }
+    // Use strlen() instead of length() because the strings taken from debug_ip_layout
+    // are always 128 in length, where the end is full of null characters
+    max1 = std::max(strlen(aCUNamePortNames.back().first.c_str()), max1);
+    max2 = std::max(strlen(aCUNamePortNames.back().second.c_str()), max2);
+  }
+  return std::pair<size_t, size_t>(max1, max2);
+}
+
+std::pair<size_t, size_t> getStreamName (const std::vector<std::string>& aSlotNames,
+                             std::vector< std::pair<std::string, std::string> >& aStreamNames) 
+{
+  //Slotnames are of the format "Master-Slave", split them and return in separate vector
+  //return max length of the Master and Slave port names
+  size_t max1 = 0, max2 = 0;
+  for (auto &s: aSlotNames) {
+    size_t found;
+    found = s.find(IP_LAYOUT_SEP, 0);
+    if (found != std::string::npos)
+      aStreamNames.emplace_back(s.substr(0, found), s.substr(found+1));
+    else
+      aStreamNames.emplace_back("Unknown", "Unknown");
+    max1 = std::max(aStreamNames.back().first.length(), max1);
+    max2 = std::max(aStreamNames.back().second.length(), max2);
+  }
+  return std::pair<size_t, size_t>(max1, max2);
+}
+
+
+int readAIMCounters(xclDeviceHandle hdl, debug_ip_layout* map) 
+{
+  std::vector<std::string> slotNames;
+  uint32_t count = getDebugIpData(map, AXI_MM_MONITOR, nullptr, &slotNames);
+  if(!count) {
+    std::cout << "ERROR: AXI Interface Monitor IP does not exist on the platform" << std::endl;
+    return 0;
+  }
+
+  xclDebugCountersResults debugResults = {0};
+  std::vector< std::pair<std::string, std::string> > cuNameportNames;
+  std::pair<size_t, size_t> widths = getCUNamePortName(slotNames, cuNameportNames);
+  xclDebugReadIPStatus(hdl, XCL_DEBUG_READ_TYPE_AIM, &debugResults);
+
+  std::cout << "AXI Interface Monitor Counters\n";
+  auto col1 = std::max(widths.first, strlen("Region or CU")) + 4;
+  auto col2 = std::max(widths.second, strlen("Type or Port"));
+
+  std::streamsize ss = std::cout.precision();
+  std::ios_base::fmtflags f(std::cout.flags());
```

Comment:
It doesn't look like 'ss' is used.  
Replace line 164 with (include core/common/utils.h):
`auto iosguard = xrt_core::utils::ios_restore(std::cout);`
and remove line 203

Reply:
ss is used in line 194. I will change the fmtflags use.